app/helpers/checkouts_helper.rb

```diff
@@ -0,0 +1,60 @@
+module CheckoutsHelper
+  def active_step(current_step)
+    'active' if current_step == step
+  end
+
+  def address_error?(field:, tag:)
+    return @billing.errors.include?(field) if @billing && tag == 'billing'
+    @shipping.errors.include?(field) if @shipping && tag == 'shipping'
+  end
+
+  def address_error_message(field:, tag:)
+    return @billing.errors[field].to_sentence if @billing && tag == 'billing'
+    @shipping.errors[field].to_sentence if @shipping && tag == 'shipping'
+  end
+
+  def address_saved_value(field:, tag:)
+    order_address_field(tag, field) || user_address_field(tag, field) || inputed_address_field(tag, field)
+  end
+
+  def checked_delivery?(delivery_id)
+    current_order.delivery_id == delivery_id
+  end
+
+  def card_error?(field)
+    @credit_card.errors.include?(field) if @credit_card
+  end
+
+  def card_error_message(field)
+    @credit_card.errors[field].to_sentence if @credit_card
+  end
+
+  def card_saved_value(field)
+    current_order.credit_card.try(field) || @credit_card.try(field)
+  end
+
+  def client_name(address)
+    address.first_name + ' ' + address.last_name
+  end
+
+  def secret_card_number(number)
+    '*** *** *** ' + number[-4, 4]
```

Comment:
Save all numbers of credit card to database is bad practice. Please save only last 4 digets.